### DISCUSSION TITLE
Restoring backup logs via extents

### DISCUSSION THREAD

**Discussion by Tthongsy** [datalust/seq-tickets#1328] [Help] [answered]
Hello,

Is there a way to restore backup logs via extents folder?

**Comment by nblumhardt** [comment, marked as answer]
Hi @Tthongsy,

Yes, with the Seq service completely stopped, it is possible to copy folders out of `Extents/` and later, also with the service stopped, to move them back in again.

It's important to note that the folders can't be merged when restoring: if a particular folder already exists under `Extents/` when restoring, then it must be deleted before the earlier copy of the same folder can be moved back in.

However, `Extents/` no longer exists in Seq; 2020.1 and later versions use `Stream/`, and this has a different layout. It's possible to do the same thing with `Stream/`, _but_ - the stream folder is not broken up into 7-day subfolders, so the entire log stream must be backed up/restored as a single unit.

This might be okay in some limited scenarios, but archiving to newline-delimited JSON with _Seq.App.JsonArchive_ is the preferred and recommended approach for long-term archival storage.

The newline-delimited JSON files produced by _Seq.App.JsonArchive_ can be individually imported back into Seq using `seqcli import --json -i <filename>`.

HTH,
Nick

**Reply by Tthongsy** [reply]
Thank you!
